Code/Framework/AzCore/AzCore/Settings/SettingsRegistryImpl.cpp

```diff
@@ -1382,12 +1445,12 @@ namespace AZ
 
         ScopedMergeEvent scopedMergeEvent(*this, { path, rootKey });
         JsonSerializationResult::ResultCode mergeResult(JsonSerializationResult::Tasks::Merge);
-        auto anchorType = Type::NoType;
+        SettingsType anchorType;
         if (rootKey.empty())
         {
             AZStd::scoped_lock lock(m_settingMutex);
             mergeResult = JsonSerialization::ApplyPatch(m_settings, m_settings.GetAllocator(), jsonPatch, mergeApproach, applyPatchSettings);
-            anchorType = SettingsRegistryImplInternal::RapidjsonToSettingsRegistryType(m_settings);
+            anchorType = GetTypeNoLock(rootKey);
```

Comment:
What's the effect of this change because it's now passing in empty rootKey instead of m_settings? Do they both resolve to the root because the key is empty?

Reply:
Yes, the `rootKey` corresponds to the root of the JSON document, which is `m_settings`.
So this grabbing the type of the JSON root which in all cases is going to be JSON object